client/src/cmdlfem4x70.c

```diff
@@ -40,100 +98,194 @@ static void fill_buffer_prng_bytes(void* buffer, size_t byte_count) {
+    PrintAndLogEx(INFO, "");
 
+    PrintAndLogEx(INFO, "Tag ID:    %02X %02X %02X %02X", data->Raw[7], data->Raw[6], data->Raw[5], data->Raw[4]);
+    PrintAndLogEx(INFO, "Lockbit 0: %d", (data->Raw[3] & LOCKBIT_0) ? 1 : 0);
+    PrintAndLogEx(INFO, "Lockbit 1: %d", (data->Raw[3] & LOCKBIT_1) ? 1 : 0);
+    PrintAndLogEx(INFO, "Tag is %s.", (data->Raw[3] & LOCKBIT_0) ? _RED_("LOCKED") : _GREEN_("UNLOCKED"));
     PrintAndLogEx(INFO, "");
-    PrintAndLogEx(INFO, "Tag ID:    %02X %02X %02X %02X", data[7], data[6], data[5], data[4]);
-    PrintAndLogEx(INFO, "Lockbit 0: %d", (data[3] & LOCKBIT_0) ? 1 : 0);
-    PrintAndLogEx(INFO, "Lockbit 1: %d", (data[3] & LOCKBIT_1) ? 1 : 0);
-    PrintAndLogEx(INFO, "Tag is %s.", (data[3] & LOCKBIT_0) ? _RED_("LOCKED") : _GREEN_("UNLOCKED"));
-    PrintAndLogEx(NORMAL, "");
 
+    PrintAndLogEx(NORMAL, "");
 }
 
-// Note: arm source has a function with same name ... different signature.
-static int em4x70_info(void) {
+// Helper functions to simplify building more complex interactions.
+// (without creating fake `const char * Cmd` arrays.)
+// These functions use 100% explicit arguments.
+// If a parameter is an output parameter, the name ends with `_out`.
 
-    em4x70_data_t edata = {
-        .parity = false // TODO: try both? or default to true
-    };
+static int get_em4x70_info(em4x70_cmd_helper_options opts, em4x70_tag_info * data_out) {
+
+    memset(data_out, 0, sizeof(em4x70_tag_info));
 
+    em4x70_data_t edata = { .parity = opts.use_parity };
     clearCommandBuffer();
-    SendCommandNG(CMD_LF_EM4X70_INFO, (uint8_t *)&edata, sizeof(edata));
+    SendCommandNG(CMD_LF_EM4X70_INFO, (uint8_t *)&edata, sizeof(em4x70_data_t));
 
     PacketResponseNG resp;
     if (!WaitForResponseTimeout(CMD_LF_EM4X70_INFO, &resp, TIMEOUT)) {
-        PrintAndLogEx(WARNING, "(em4x70) Timeout while waiting for reply.");
+        if (!opts.disable_error_output) {
+            PrintAndLogEx(WARNING, "(em4x70) Timeout while waiting for reply.");
+        }
+        return PM3_ETIMEOUT;
+    }
+    if (resp.status) {
+        memcpy(data_out, resp.data.asBytes, sizeof(em4x70_tag_info));
+        return PM3_SUCCESS;
+    }
+    return PM3_ESOFT;
+}
+static int write_em4x70_block(em4x70_cmd_helper_options opts, em4x70_tag_info* data_out, int block, uint16_t value) {
+
+    memset(data_out, 0, sizeof(em4x70_tag_info));
+
+    em4x70_data_t etd = {0};
+    etd.address = (uint8_t) block;
+    etd.word = value; // ISSUE: Presumes client and target are both little-endian
+    etd.parity = opts.use_parity;
+
+    clearCommandBuffer();
+    SendCommandNG(CMD_LF_EM4X70_WRITE, (uint8_t *)&etd, sizeof(etd));
+
+    PacketResponseNG resp;
+    if (!WaitForResponseTimeout(CMD_LF_EM4X70_WRITE, &resp, TIMEOUT)) {
+        if (!opts.disable_error_output) {
+            PrintAndLogEx(WARNING, "Timeout while waiting for reply.");
+        }
         return PM3_ETIMEOUT;
     }
 
     if (resp.status) {
```

Comment:
...  the PM3_SUCCESS is defined as 0.   If you are pushing your own status codes,  please not. 
To make the understanding of the status code,   always use a comparision,   which allows for easier understanding of what is expected outcome from the device.

Reply:
This is definitely not creating new status codes.
Rather, the `opts` parameter has a different purpose:  Allowing code re-use without forcing specific outputs to be sent.   For example, being able to re-use the `auth` helper function without a gazillion error messages, if wanting to try a brute force.  Or, periodically looping with a check for a tag, where you want to silently ignore a tag not being present.

This PR was intended to make a cleaner split between client command line parsing, and functions that perform a specific interaction with the firmware.   There are no new error codes.

Does that help explain some of the logic here and address your concern about new error codes?